serialization/src/main/kotlin/Encoder.kt

```diff
@@ -0,0 +1,61 @@
+package com.juul.mcumgr.serialization
+
+import com.fasterxml.jackson.databind.node.ObjectNode
+import com.juul.mcumgr.message.Format
+import com.juul.mcumgr.message.Request
+import okio.Buffer
+import okio.BufferedSink
+
+fun Request.encode(format: Format, sequenceNumber: Int = 0): ByteArray {
+    val header = Header(
+        operation.value,
+        group.value,
+        command.value,
+        0,
+        sequenceNumber,
+        0
+    )
+    val payload: ObjectNode = cbor.valueToTree(this)
+    val message = Message(header, payload)
+    return message.encode(format)
+}
+
+fun Message.encode(format: Format): ByteArray =
+    when (format) {
+        Format.STANDARD -> encodeStandard()
+        Format.COAP -> encodeCoap()
+    }
+
+private fun Message.encodeStandard(): ByteArray =
+    Buffer().apply {
+        // Set the header's length to the size of the payload.
+        val payloadBytes = encodePayload()
+        val lengthSetHeader = header.copy(length = payloadBytes.size)
+        encodeHeader(lengthSetHeader)
+        write(payloadBytes)
+    }.readByteArray()
+
+private fun Message.encodeCoap(): ByteArray {
```

Comment:
Is this encoding CoAP or CBOR?

Reply:
Same as the other comment, this is the "Coap" mcumgr message format for CoAP based transports.  

The other name would be SMP vs OMP, which is probably better now that I think about it. as that is what the other mcumgr libraries use.